Appunte/forms.py

```diff
@@ -20,18 +20,63 @@ class Meta:
         fields = ('email', 'username', 'rol')
 
 
-class RegistrationForm(UserCreationForm):
-    email = forms.EmailField(required=True)
+class RegistrationForm(forms.Form):
+
     class Meta:
         model = CustomUser
         fields = ('email', 'username', 'password')
 
-    def save(self, commit=True):
-        CustomUser = super(RegistrationForm, self).save(commit=False)
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    username = forms.CharField(
+        label='Nombre de Usuario',
+        max_length=30,
+    )
+
+    email = forms.EmailField(
+        label='Correo',
+    )
+
+    password = forms.CharField(
+        label='Contraseña',
+    )
+
+
+    def clean_username(self):
+        cleaned_data = super(RegistrationForm, self).clean()
+        username = cleaned_data['username']
+        if CustomUser.objects.filter(username=username):
+            self.add_error(
+                'username',
+                forms.ValidationError("User Exists")
+                )
+        return username
+
+    def clean_email(self):
+        cleaned_data = super(RegistrationForm, self).clean()
+        email = cleaned_data['email']
+        if CustomUser.objects.filter(email=email):
+            self.add_error(
+                'email',
+                forms.ValidationError(
+                  "Mail exists")
+                )
+        return email
```

Comment:
No estoy seguro cuando se llaman estos métodos. Creo que forms debe llamar a todos métodos que comienzan con "clean" cuando se llama el clean() en la linea 71

Reply:
Got it